R/derive_advs_params.R

```diff
@@ -0,0 +1,633 @@
+                                  get_unit_expr) {
+  assert_vars(by_vars)
+  assert_data_frame(dataset, required_vars = exprs(!!!by_vars, PARAMCD, AVAL))
+  assert_character_scalar(wstcir_code)
+  assert_character_scalar(hipcir_code)
+  assert_varval_list(set_values_to, required_elements = "PARAMCD")
+  assert_param_does_not_exist(dataset, set_values_to$PARAMCD)
+  filter <- assert_filter_cond(enexpr(filter), optional = TRUE)
+  get_unit_expr <- assert_expr(enexpr(get_unit_expr))
+
+  units_supported <- names(get_conv_factors_all()[["length"]])
+
+  assert_unit(
+    dataset,
+    param = wstcir_code,
+    required_unit = units_supported,
+    get_unit_expr = !!get_unit_expr
+  )
+
+  assert_unit(
+    dataset,
+    param = hipcir_code,
+    required_unit = units_supported,
+    get_unit_expr = !!get_unit_expr
+  )
+
+  derive_param_ratio(
+    dataset,
+    filter = !!filter,
+    numerator_code = wstcir_code,
+    denominator_code = hipcir_code,
+    by_vars = by_vars,
+    set_values_to = set_values_to,
+    get_unit_expr = !!get_unit_expr
+  )
+}
+
+#' Adds a Parameter for Waist to Height Ratio
+#'
+#' @description Adds a parameter for Waist to Height Ratio using Waist Circumference and Height
+#' for each by group (e.g., subject and visit) where the source parameters are available.
+#'
+#' **Note:** This is a wrapper function for the more generic [`admiral::derive_param_computed()`].
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `by_vars` argument are expected to be in the dataset.
+#'   `PARAMCD`, and `AVAL` are expected as well.
+#'
+#'   The variable specified by `by_vars` and `PARAMCD` must be a unique key of the input dataset
+#'   after restricting it by the filter condition (`filter` argument) and to the parameters
+#'   specified by `wstcir_code` and `height_code`.
+#'
+#' @param wstcir_code Waist Circumference parameter code
+#'
+#'   The observations where `PARAMCD` equals the specified value are considered
+#'   as the Waist Circumference.
+#'
+#'   *Permitted Values:* character value
+#'
+#' @param height_code Height parameter code
+#'
+#'   The observations where `PARAMCD` equals the specified value are considered as the Height.
+#'
+#'   *Permitted Values:* character value
+#'
+#' @param constant_by_vars By variables for when Height is constant
+#'
+#'   When Height is constant, the Height parameters (measured only once) are merged
+#'   to the other parameters using the specified variables.
+#'
+#'   If Height is constant (e.g. only measured once at screening or baseline) then use
+#'   `constant_by_vars` to select the subject-level variable to merge on (e.g. `USUBJID`).
+#'   This will produce Waist to Height Ratio at all visits where Waist Circumference is measured.
+#'   Otherwise it will only be calculated at visits with both Height and Waist Circumference
+#'   collected.
+#'
+#'   *Permitted Values*: list of variables created by `exprs()`, e.g. `exprs(USUBJID, VISIT)`
```

Comment:
I am assuming this comes from admiral? Would it not in reality be list of symbols? Or maybe add something indicating we use data masking. Here is an example from `ggplot2::aes()`

![image](https://github.com/user-attachments/assets/47253dfa-d396-4136-8c21-955952b735cf)

Reply:
It's consistent with other admiral functions.